### PR TITLE
Doc

### PR DESCRIPTION
**Pull request by heavenboy8**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,17 +1,84 @@
...
+```
+
+This image:
+- contains the fully configured IMA
+- contains the fully configured SELinux 
+- disables the auto-update (to avoid any modification of the Cosmian VM after having snapshoted it)
+- contains the fully configured `cosmian_vm_agent` and a nginx to redirect queries to it
+- contains a script `cosmian_vm_post_install.sh` to finalize the configuration of the Cosmian VM after installing the image on a fresh environment. It will setup the SSL certificate and finalize the configuration of the nginx & `cosmian_vm_agent`
+
+You can start/restart/stop the Cosmian VM Agent as follow:
```
Comment (ThibsG) [inline README.md:56]: maybe we can add information about what is `cosmian_vm_agent` in one phrase?

File: README.md [hunk 2]
```diff
@@ -1,17 +1,84 @@
...
+
+```sh
+$ supervisorctl start cosmian_vm_agent
+$ supervisorctl restart cosmian_vm_agent
+$ supervisorctl stop cosmian_vm_agent
+ ```
+
 ## Compile
```
Comment (ThibsG) [inline README.md:64]:
```suggestion
## Compile and run tests
```

File: README.md [hunk 3]
```diff
@@ -1,17 +1,84 @@
...
 
 ```sh
 $ cargo build
+$ cargo test
 ```
 
 ## Usage
 
-On a SEV host:
+On a configured SEV/TDX host, create and edit the `cosmian_vm_agent` configuration file located at `/etc/cosmian_vm/agent.conf`.
```
Comment (ThibsG) [inline README.md:73]:
```suggestion
On a configured SEV/TDX host, create and edit the `cosmian_vm_agent` configuration file located at `/etc/cosmian_vm/agent.toml`. 
```

File: README.md [hunk 4]
```diff
@@ -1,17 +1,84 @@
...
+```toml
+[agent]
+pem_certificate = "/etc/letsencrypt/live/cosmianvm.cosmian.dev/cert.pem"
+```
+
+Then, run the  `cosmian_vm_agent` as follow:
 
 ```sh  
-$ sudo COSMIAN_VM_AGENT_CERTIFICATE=/etc/letsencrypt/live/cosmianvm.cosmian.dev/cert.pem ./cosmian_vm_agent
+$ sudo ./cosmian_vm_agent
```
Comment (ThibsG) [inline README.md:84]: it is not run as a service (via `supervisorctl`)?

File: README.md [hunk 5]
```diff
@@ -27,3 +94,46 @@ $ cosmian_vm snapshot --url https://cosmianvm.cosmian.dev
 ```sh
 $ cosmian_vm verify --url https://cosmianvm.cosmian.dev --snapshot cosmian_vm.snapshot  
 ```
+
+## Provide secrets
+
+You can also provide a secret file to an application running inside the Cosmian VM. 
+
+Prio to send the secrets, you should have configured the  `app` section in the `agent.conf` as follow:
```
Comment (ThibsG) [inline README.md:105]:
```suggestion
Prior to send the secrets, you should have configured the  `app` section in the `agent.toml` as follow:
```

File: README.md [hunk 6]
```diff
@@ -27,3 +94,46 @@ $ cosmian_vm snapshot --url https://cosmianvm.cosmian.dev
...
+[agent]
+pem_certificate = "/etc/letsencrypt/live/cosmianvm.cosmian.dev/cert.pem"
+
+[app]
+service_app_name = "cosmian_helloworld"
+decrypted_folder = "/mnt/cosmian_vm/data"
+encrypted_secret_app_conf = "/etc/cosmian_vm/app_secrets.json"
+```
+
+In that example, `cosmian_helloworld` is the name of the application (as a `supervisor` service).
```
Comment (ThibsG) [inline README.md:117]: note that for now, `helloworld-service` project is private
```suggestion
In that example, [`cosmian_helloworld`](https://github.com/Cosmian/helloworld-service) is the name of the application (as a `supervisor` service). 
```

File: README.md [hunk 7]
```diff
@@ -27,3 +94,46 @@ $ cosmian_vm snapshot --url https://cosmianvm.cosmian.dev
...
+
+[app]
+service_app_name = "cosmian_helloworld"
+decrypted_folder = "/mnt/cosmian_vm/data"
+encrypted_secret_app_conf = "/etc/cosmian_vm/app_secrets.json"
+```
+
+In that example, `cosmian_helloworld` is the name of the application (as a `supervisor` service). 
+- `decrypted_folder` stands for the directory where the application expects to find its decrypted configuration file (which should be located into an encrypted RAMFS)
+- `encrypted_secret_app_conf` stands for the location where `cosmian_vm_agent` store the application configuration encrypted
```
Comment (ThibsG) [inline README.md:119]:
```suggestion
- `encrypted_secret_app_conf` stands for the location where `cosmian_vm_agent` stores the application configuration encrypted
```

File: README.md [hunk 8]
```diff
@@ -27,3 +94,46 @@ $ cosmian_vm snapshot --url https://cosmianvm.cosmian.dev
...
+```sh
+$ cosmian_vm app init --url https://cosmianvm.cosmian.dev --conf secrets.json
+```
+
+The configuration file can be anything the application expects. Here, a json file. 
+
+The configuration file will be encrypted by the `cosmian_vm_agent` and stored encrypted in the value set in `encrypted_secret_app_conf`. 
+A decrypted version of this file will be saved in the value set in `decrypted_folder`. The `init` subcommand will finally start the application identified in `service_app_name` field. 
+
+The key used to encrypt this file will be prompted in the output of the `init` command if you don't provide it as an args of the command using `--key`.
```
Comment (ThibsG) [inline README.md:132]:
```suggestion
The key used to encrypt this file will be prompted in the output of the `init` command if you don't provide it as an arg of the command using `--key`.
```

**Review by phochard — APPROVED** [review]

**Review by ThibsG — APPROVED** [review]

**Pull request merged**